pkg/azuredisk/nodeserver.go

```diff
@@ -324,22 +324,32 @@ func (d *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (
 		Segments: map[string]string{topologyKey: ""},
 	}
 
+	var failureDomainFromLabels, instanceTypeFromLabels string
+	var err error
+
 	if d.supportZone {
-		var (
-			zone      cloudprovider.Zone
-			zoneError error
-		)
-		if runtime.GOOS == "windows" && (!d.cloud.UseInstanceMetadata || d.cloud.Metadata == nil) {
-			zone, zoneError = d.cloud.VMSet.GetZoneByNodeName(d.NodeID)
+		var zone cloudprovider.Zone
```

Comment:
I believe that the code is correct however I think it's a little bit hard to read, it might be a good idea to refactor this into a function so that we can return the zone as soon as it's available (same thing for the instance type), I don't think I have a concern about the logic though.

Thanks for making the change.